### PR TITLE
Feature/setting

### PR DESCRIPTION
**Pull request by Npwskp**
CRUD setting

### CODE REVIEW

File: src/setting/setting.service.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+@Injectable()
+export class SettingService {
+  constructor(
+    @InjectModel(SettingSchema)
+    private readonly settingModel: ReturnModelType<typeof SettingSchema>,
+  ) {}
+
+  async getSettings() {
+    const settings = await this.settingModel.find()[0];
```
Comment (ishiasaka) [inline src/setting/setting.service.ts]: You can use findOne without condition to get same result.
Don't forget to .exec()

File: src/setting/setting.service.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  async createSettings(name: string, logo: string) {
+    const settings = await this.settingModel.create({
+      name,
+      logo,
+    });
+    return settings;
+  }
+
+  async updateSettings(settingDto: SettingDto) {
+    const settings_id = await this.settingModel.find()[0]._id;
```
Comment (ishiasaka) [inline src/setting/setting.service.ts]: You can use `this.getSettings` from above.

File: src/setting/setting.service.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+@Injectable()
+export class SettingService {
+  constructor(
+    @InjectModel(SettingSchema)
+    private readonly settingModel: ReturnModelType<typeof SettingSchema>,
+  ) {}
+
+  async getSettings() {
+    const settings = await this.settingModel.findOne();
```
Comment (vjumpkung) [inline src/setting/setting.service.ts]: add `.exec()`

File: src/setting/setting.service.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    const settings_id = await this.getSettings();
+    const settings = await this.settingModel.findOneAndUpdate(
+      { _id: settings_id },
+      {
+        $set: settingDto,
+      },
+      {
+        new: true,
+      },
+    );
```
Comment (vjumpkung) [inline src/setting/setting.service.ts]: add `.exec()`

**Review by vjumpkung — APPROVED** [review]
LGTM but คนอื่น LGTM มั้ย 55555555555

**Pull request merged**